src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -50,14 +50,16 @@ int find_output(UniValue obj, int n) {
 }
 
 AsyncRPCOperation_sendmany::AsyncRPCOperation_sendmany(
+        CMutableTransaction contextualTx,
         std::string fromAddress,
         std::vector<SendManyRecipient> tOutputs,
         std::vector<SendManyRecipient> zOutputs,
         int minDepth,
         CAmount fee,
         UniValue contextInfo) :
-        fromaddress_(fromAddress), t_outputs_(tOutputs), z_outputs_(zOutputs), mindepth_(minDepth), fee_(fee), contextinfo_(contextInfo)
+        tx_(contextualTx), fromaddress_(fromAddress), t_outputs_(tOutputs), z_outputs_(zOutputs), mindepth_(minDepth), fee_(fee), contextinfo_(contextInfo)
 {
+    assert(contextualTx.nVersion >= 2);  // transaction format version must support vjoinsplit
```

Comment:
A side-effect of this is that `z_sendmany` no longer creates v1 transactions for `t->t`. This should be fine though, as anyone who has made the effort to use `z_sendmany` should already support parsing v2 transactions.

Reply:
Is this addition related to NU0?

Reply:
This is technically violating a SHOULD in the spec: "Software that creates transactions SHOULD use version 1 for transactions with no JoinSplit descriptions."

Shall I remove this from the spec?